ee/gowrapper/goroutine.go

```diff
@@ -0,0 +1,35 @@
+package gowrapper
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/pkg/errors"
+)
+
+// Go is a thin wrapper around `go func()` that ensures we log panics
+// and then handle them appropriately. `onPanic` defines the behavior
+// that should happen after recovering from a panic.
+func Go(slogger *slog.Logger, goroutine func(), onPanic func(r any)) {
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				slogger.Log(context.TODO(), slog.LevelError,
```

Comment:
Should we pass in ctx?

Reply:
I was waffling on whether it felt worth it or not -- happy to add it if you'd like